continuous_integration/flake8_diff.sh

```diff
@@ -39,4 +68,5 @@ echo '--------------------------------------------------------------------------
 
 # Conservative approach: diff without context so that code that was
 # not changed does not create failures
-git diff --unified=0 $COMMIT | flake8 --diff && echo -e "No problem detected by flake8\n"
+git diff --unified=0 $COMMIT | flake8 --diff --show-source
```

Comment:
This is something I haven't figured out fully, but somehow in the bash script, set -e did not have the expected effect with &&. Basically it would not stop if there was an error. Note the behaviour is exactly the same, because flake8 exits with a non zero exit code if there are PEP8 violations.